app/src/main/kotlin/com/wire/android/ui/home/AppSyncViewModel.kt

```diff
@@ -83,6 +85,7 @@ class AppSyncViewModel @Inject constructor(
             listOf(
                 viewModelScope.launch(dispatcher.io()) { syncCertificateRevocationListUseCase() },
                 viewModelScope.launch(dispatcher.io()) { featureFlagsSyncWorker.execute() },
+                viewModelScope.launch(dispatcher.io()) { mlsPublicKeysSyncWorker.executeImmediately() },
```

Comment:
I am pretty sure this whole block of `runSyncTasks()` here was just a quick hack and now it is getting expanded further.

App knows too much about this. 

We should either:
A. Launch `mlsPublicKeysSyncWorker` when `UserSessionScope` starts, and have it wait for Sync to be LIVE before doing anything else. Once LIVE it will do its thing, be it on the background or not. And it can reschedule itself to repeat every 24h if still running.
B. Create a `ForegroundWorker` of some sorts in Kalium, so the app would call it and Kalium would take over.

Reply:
The documentation states that it should be called each time the user puts the app in the foreground, so running it each time the app becomes LIVE is not exactly the same.
And it actually waits until live when being executed, this logic is implemented already inside this worker, just like for feature flags.
About B idea, I don't know if I get it correctly but the only change would be that the app starts this single `ForegroundWorker` instead of starting each one separately?

Reply:
I believe that after adjusting this PR to use new `ForegroundActionsUseCase` this can be resolved 😄